crates/nu-command/src/filesystem/cd.rs

```diff
@@ -246,14 +247,43 @@ fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
             use std::os::unix::fs::MetadataExt;
             let bits = metadata.mode();
             let has_bit = |bit| bits & bit == bit;
-            let current_user = users::get_current_uid();
-            if current_user == 0 {
+            let current_user_uid = users::get_current_uid();
+            if current_user_uid == 0 {
                 return PermissionResult::PermissionOk;
             }
-            let current_group = users::get_current_gid();
+
+            let current_user_gid = users::get_current_gid();
+            let mut current_user_groups = match users::get_current_username() {
+                None => {
+                    return PermissionResult::PermissionDenied("Unable to get current user name")
+                }
+                Some(name) => match users::get_user_groups(&name, current_user_gid) {
+                    None => {
+                        return PermissionResult::PermissionDenied(
+                            "Unable to get current user groups",
+                        )
+                    }
+                    Some(groups) => {
+                        let mut iter = groups.into_iter();
+
+                        // Fixes https://github.com/ogham/rust-users/issues/44
+                        // If a user isn't in more than one group then this fix won't work,
```

Comment:
I think the comment line is wrong, the fix won't work when the user have only two group, one user group, one root group, then it still can't go to the directory

Reply:
I'm pretty sure it will since if the user is in lets say `user,root` the returned vec would be `user,root,root` so it would still remove the additional root

Reply:
Hmm...here is what I get:
```
/tmp〉id -Gn
tmp_user root
/tmp〉cd permission_test/
debug groups: [Group(1002, tmp_user), Group(0, root)]
```

Reply:
Thats correct no? The user has the user group and root and its removed the additional root from the list so it isn't `1022, 0, 0`

Reply:
No... I've added some debug message to your code:
```
println!("before: {:?}", groups);
if groups.len() >= 2 {
    groups.pop();
}

println!("after: {:?}", groups);
groups
```

Then here is my running result:
```
/tmp〉ls -lD pp | select name type uid group
╭───┬──────┬──────┬──────┬───────╮
│ # │ name │ type │ uid  │ group │
├───┼──────┼──────┼──────┼───────┤
│ 0 │ pp   │ dir  │ root │ root  │
╰───┴──────┴──────┴──────┴───────╯
/tmp〉id -Gn
tmp_user root
/tmp〉cd pp
before: [Group(1002, tmp_user), Group(0, root)]
after: [Group(1002, tmp_user)]
Error: nu::shell::io_error (link)

  × I/O error
  help: Cannot change directory to /tmp/pp: You are neither the owner, in the group, nor the super user and do not have
        permission
```

As you can see, if the user have only two groups `tmp_user`, `root`, `users::get_user_groups` returned vec would be `tmp_user`, `root`; without an extra `root`..

Reply:
Hmm, well that does complicate it, let me see what i can do

Reply:
Newest commit fixes that with some more explaining of what's going on.